src/connectors/range-slider/connectRangeSlider.js

```diff
@@ -76,108 +80,146 @@ export default function connectRangeSlider(renderFn) {
+
+          const newNextMax =
+            !isMaxBounds && rangeMax === nextMax ? undefined : nextMax;
+
+          if (min !== newNextMin || max !== newNextMax) {
+            helper.clearRefinements();
+
+            const isValidMinInput = _isFinite(newNextMin);
+            const isValidMinRange = _isFinite(rangeMin);
+            const isGreatherThanRange =
+              isValidMinRange && rangeMin <= newNextMin;
+
+            if (isValidMinInput && (!isValidMinRange || isGreatherThanRange)) {
               helper.addNumericRefinement(
                 attributeName,
                 '>=',
-                formatToNumber(newValues[0])
+                formatToNumber(newNextMin)
               );
             }
 
-            const hasMax = bounds.max !== null && bounds.max !== undefined;
-            const maxValueChanged =
-              newValues[1] !== null && newValues[1] !== undefined;
+            const isValidMaxInput = _isFinite(newNextMax);
+            const isValidMaxRange = _isFinite(rangeMax);
+            const isLowerThanRange = isValidMaxRange && rangeMax >= newNextMax;
 
-            if (
-              (hasMax && maxValueChanged && bounds.max > newValues[1]) ||
-              (!hasMax && maxValueChanged)
-            ) {
+            if (isValidMaxInput && (!isValidMaxRange || isLowerThanRange)) {
               helper.addNumericRefinement(
                 attributeName,
                 '<=',
-                formatToNumber(newValues[1])
+                formatToNumber(newNextMax)
               );
             }
+
             helper.search();
           }
         };
+      },
 
-        const stats = {
-          min: userMin || null,
-          max: userMax || null,
+      getConfiguration(currentConfiguration) {
+        const configuration = {
+          disjunctiveFacets: [attributeName],
         };
-        const currentRefinement = this._getCurrentRefinement(helper);
+
+        const boundsAlreadyDefined =
+          currentConfiguration &&
+          currentConfiguration.numericRefinements &&
+          currentConfiguration.numericRefinements[attributeName] !== undefined;
+
+        const isBoundsDefined = isMinBounds || isMaxBounds;
+        const isAbleToRefine = this._isAbleToRefine(minBounds, maxBounds);
+
+        if (isBoundsDefined && !boundsAlreadyDefined && isAbleToRefine) {
+          configuration.numericRefinements = { [attributeName]: {} };
+
+          if (isMinBounds) {
+            configuration.numericRefinements[attributeName]['>='] = [minBounds];
+          }
+
+          if (isMaxBounds) {
+            configuration.numericRefinements[attributeName]['<='] = [maxBounds];
+          }
+        }
+
+        return configuration;
+      },
+
+      init({ helper, instantSearchInstance }) {
+        const stats = {};
+        const range = this._getCurrentRange(stats);
+        const { min, max } = this._getCurrentRefinement(helper);
```

Comment:
`_getCurrentRefinement` should return an array so we don't have to extract the values and create an array.